src/vulkan/buffer_descriptor.cc

```diff
@@ -123,5 +127,21 @@ std::vector<Resource*> BufferDescriptor::GetResources() {
   return ret;
 }
 
+bool BufferDescriptor::IsReadOnly() {
+  switch (type_) {
+    case DescriptorType::kUniformBuffer:
+    case DescriptorType::kUniformBufferDynamic:
+    case DescriptorType::kUniformTexelBuffer:
+      return true;
+    case DescriptorType::kStorageBuffer:
+    case DescriptorType::kStorageBufferDynamic:
+    case DescriptorType::kStorageTexelBuffer:
+      return false;
+    default:
+      assert(false && "Unexpected descriptor type");
+      return false;
+  }
+}
```

Comment:
Perhaps this should be implemented in `Descriptor` for all types? We could add assertions elsewhere (constructors?) so that the `type_` must make sense for the subclass.

Reply:
I would move this to BufferBackedDescriptor instead as it doesn't make any sense for samplers, and it's only used for subclasses of BufferBackedDescriptor.